src/ipc/ipc4/helper.c

```diff
@@ -478,13 +478,18 @@ int ipc_comp_disconnect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	 * pipelines need to be on the same core.
 	 */
 	irq_local_disable(flags);
-	pipeline_disconnect(src, buffer, PPL_CONN_DIR_COMP_TO_BUFFER);
-	pipeline_disconnect(sink, buffer, PPL_CONN_DIR_BUFFER_TO_COMP);
+	if (src->pipeline->status < COMP_STATE_ACTIVE)
```

Comment:
Can you circle back and add inline comments here for this so it's easy to understand.

Reply:
sure, thanks